modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js

```diff
@@ -78,17 +56,39 @@ const getDelimiter = (dateFormat) => {
 	return dateDelimiter;
 };
 
+const getLocaleDateFormat = (format = 'L') => {
+	return moment.localeData().longDateFormat(format);
+};
+
+const getMaskByDateFormat = (format) => {
+	const mask = [];
+
+	for (let i = 0; i < format.length; i++) {
+		if (/[a-z]/i.test(format[i])) {
+			mask.push(/\d/);
+		}
+		else {
+			mask.push(`${format[i]}`);
+		}
+	}
+
+	return mask;
+};
+
 const getDateFormat = () => {
-	const dateFormat = Liferay.AUI.getDateFormat();
-	const dateDelimiter = getDelimiter(dateFormat);
+	const dateMask = getLocaleDateFormat();
+	const inputMask = getMaskByDateFormat(dateMask);
+	const dateDelimiter = getDelimiter(inputMask);
 
 	return {
-		dateMask: getDateMask(dateFormat, dateDelimiter),
-		inputMask: getInputMask(dateFormat, dateDelimiter),
+		dateMask: getDateMask(dateMask, dateDelimiter),
+		inputMask,
 	};
 };
 
-const transformToDate = (date) => {
+const transformToDate = (date, locale) => {
+	moment.locale(locale);
```

Comment:
@alinedoleron, shouldn't this line be inside the `if`? After all, the moment is only used inside of it.

Additionally, can we do something like `moment(date).locale(locale).toDate();`?